libs/coin-polkadot/src/bridge/prepareTransaction.test.ts

```diff
@@ -0,0 +1,48 @@
+import BigNumber from "bignumber.js";
+import { createFixtureAccount, createFixtureTransaction } from "../types/model.fixture";
+import prepareTransaction from "./prepareTransaction";
+import { faker } from "@faker-js/faker";
+
+const mockGetEstimatedFees = jest.fn();
+jest.mock("../logic", () => ({
+  getEstimatedFees: () => mockGetEstimatedFees(),
+}));
+
+describe("prepareTransaction", () => {
+  afterEach(() => {
+    mockGetEstimatedFees.mockClear();
+  });
+
+  it("returns a new Transaction with new fees", async () => {
+    // Given
```

Comment:
We could use this Given When Then pattern (Gerkhin) as describe and beforeEach clauses, this is what they are here for ! In my experience though, it gets heavy quickly and the only important thing is a high branch coverage of your tested method

Reply:
Those are only comments to clearly separate data needed from what you test and expect.

However I agree they can tend to be heavier, but it highlights the heavy dependency of what you want to test.

Reply:
We discussed this point in our huddle, makes sense with multiple contexts to add describe clauses